### ISSUE TITLE
Webhoops Paypaments problem -orders on Prestashop are not created when customer will not click back to shop, or close the payment window

### ISSUE THREAD

**Issue by adamus007p** [202-ecommerce/stripe_official#80]
TL;DR When customer make a payment using non card method (Sofort, P24) and will not click back to shop there is no order on Prestashop side.

I had from time to time customers who paid me money.
I can see it on Stripe dashboard but there was no order on Prestashop side.
No order, no error, nothing.

Some customers are too fast and will not wait or click back to shop and there is problem that there will be no orders on Prestashop.

I think that this is important problem.

To reproduce the order
1. start any order (not using a card)
2. after successful order do not click back to shop or do not wait, just close the window
3. result: on stripe side you will get a money, but there will be no order on prestashop side

I had a many customers like that as I finally figure out that is the problem.

**Comment by mribeiro202** [comment]
Hello @adamus007p ,

Thank you for your issue, we are currently working on this problem.
I recommend that you contact the support team so that they can provide you with help.
Here is the link to access it ==> https://addons.prestashop.com/en/contact-us?id_product=24922

Have a nice day

**Comment by adamus007p** [comment]
@mribeiro202 are you from 202-ecommerce - the developer of Stripe module?


The link is not working. You can also open a ticket on Prestashop site too.


I have report I think quite important bug.
I see that handling webhooks and errors in your module is quite poor.

I have 12 years of experience (as developer too) with prestashop so I little bit know few things ;)

Have a nice day

**Comment by testing3-apps** [comment]
I'm facing exactly the same problem with card payments too!

**Comment by testing3-apps** [comment]
I'm facing a similar problem with card payments too! The order is created, but the products remain in the cart until the customer clicks the back button, goes to the home page or refreshes the page.
I'm using Prestashop 1.6.1.24.

**Comment by adamus007p** [comment]
I have got from Stripe a fix, but I did not test it.


https://addons.prestashop.com/attachment/community/806/610bb01ee35dc.zip

if you know and you are advanced user you may test it too.

Please make a backup of database and old module

**Comment by testing3-apps** [comment]
@adamus007p Thank you so much, I'll test it today and let you know if it works.
I'm using an old version of the PayPal module and it's working perfectly fine with Prestashop 1.6.1.24.

**Comment by testing3-apps** [comment]
I just tested the provided version, unfortunately, it didn't work. I'm having the same problem, the order is created successfully and I received the confirmation email, but the products remained in the cart. they disappear when I click the back button, go to the home page, or refresh the page.

**Comment by Ashley-Butcher** [comment]
Having this same problem and it's not helping customers have confidence. Once again we switch Stripe off and move back to other payment brokers!

What we are seeing is for anything through Stripe the orders become abandoned carts, but we see the payment for the order in the Stripe dashboard but we need to "capture" them. We are doing the best we can to "capture" them and then manually convert the abandoned carts into orders but it requires talking to the customer first.

Likely related to [issue 77](https://github.com/202-ecommerce/stripe_official/issues/77) and [issue 71](https://github.com/202-ecommerce/stripe_official/issues/71)?

**Comment by adamus007p** [comment]
@Ashley-Butcher do you have any alternatives?

I report this problem to developer and when I will have an update I will write here too.

**Comment by adamus007p** [comment]
I can confirm that problem with web hooks has been fixed.
Please update to the nee version.

Of course remember to back a backup and test it in test environment.

**Comment by mribeiro202** [comment]
Hi @adamus007p 

Regarding the problem you describe for the shopping cart, sometimes the webhook response is slower than the redirection to the confirmation page. So the cart stays full and empties once the webhook sends its response.

We know this is a problem and we are working on a fix for a future version.

Since your initial problem is solved, I'll close this issue. Don't hesitate to contact us again if you need help.

Have a nice day.
